app/src/main/java/com/vultisig/wallet/ui/models/deposit/DepositFormViewModel.kt

```diff
@@ -1120,6 +1126,155 @@ internal class DepositFormViewModel @Inject constructor(
         )
     }
 
+    private suspend fun createAddCacaoPoolTransaction(): DepositTransaction {
+        val chain = chain
+            ?: throw InvalidTransactionDataException(
+                UiText.StringResource(R.string.send_error_no_address)
+            )
+
+        val address = accountsRepository.loadAddress(
+            vaultId,
+            chain
+        )
+            .first()
+        val selectedToken = address.accounts.first { it.token.isNativeToken }.token
+
+        if (selectedToken.ticker != "CACAO") {
+            throw InvalidTransactionDataException(
+                UiText.StringResource(R.string.cacaopool_only_cacao_supported)
+            )
+        }
+
+        val tokenAmount = tokenAmountFieldState.text
+            .toString()
+            .toBigDecimalOrNull()
+
+        if (tokenAmount == null || tokenAmount <= BigDecimal.ZERO) {
+            throw InvalidTransactionDataException(
+                UiText.StringResource(R.string.send_error_no_amount)
+            )
+        }
+        val tokenAmountInt =
+            tokenAmount
+                .movePointRight(selectedToken.decimal)
+                .toBigInteger()
+
+        val srcAddress = selectedToken.address
+
+        val gasFee = gasFeeRepository.getGasFee(
+            chain,
+            srcAddress
+        )
+        val memo = DepositMemo.DepositPool
+
+        val specific = blockChainSpecificRepository
+            .getSpecific(
+                chain,
+                srcAddress,
+                selectedToken,
+                gasFee,
+                isSwap = false,
+                isMaxAmountEnabled = false,
+                isDeposit = true,
+                transactionType = TransactionType.TRANSACTION_TYPE_GENERIC_CONTRACT,
```

Comment:
I believe this is wrong , because addCacao is not a smart contract call , it is just a deposit with memo